libs/terrain.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Perlin noise generated terrain abstraction.
+ */
+
+class Terrain
+{
+    /**
+     * I initialize myself.
+     * 
+     * Args:
+     *  height (number): terrain height
+     *  width (number): terrain width
+     *  resolution (number): terrain resolution
+     * 
+     * Returns:
+     *  undefined.
+     */
+    constructor(width, height, resolution)
+    {
+        // store components
+        this.coordinates = {x: 0, y: 0};
+        this.terrain = [];
+        this.seed = 0;
+
+        // store parameters
+        this.width = width;
+        this.height = height;
+        this.resolution = resolution;
+    }
+
+    /**
+     * I return my number of rows.
+     * 
+     * Returns:
+     *  _ (number): number of rows
+     */
+    get rows()
+    {
+        return this.height / this.resolution;
+    }
+
+    /**
+     * I return my number of columns.
+     * 
+     * Returns:
+     *  _ (number): number of columns
+     */
+    get columns()
+    {
+        return this.width / this.resolution;
+    }
+
+    /**
+     * I load my initial state.
+     * 
+     * Returns:
+     *  undefined.
+     */
+    setup()
+    {
+        this.setupNoise();
+    }
+
+    /**
+     * I load my initial perlin noise space state.
+     * 
+     * Returns:
+     *  undefined.
+     */
+    setupNoise()
+    {
+        // setup terrain
+        for(let row = 0; row < this.rows * 3; row++)
+        {
+            this.terrain.push(Array(this.columns).fill(0));
+        }
+    }
+
+    /**
+     * I update the terrain state in one step.
+     * 
+     * Returns:
+     *  undefined.
+     */
+    update()
+    {
+        // update terrain heights
+        this.coordinates.y = this.seed;
+        for(let row = 0; row < this.rows; row++)
+        {
+            this.coordinates.x = 0;
+            for(let column = 0; column < this.columns; column++)
+            {
+                this.terrain[column][row] = map(noise(this.coordinates.x, this.coordinates.y), 0, 1, -100, 100);
+                this.coordinates.x += 0.2;
+            }
+            this.coordinates.y += 0.2;
+        }
+    }
+
+    /**
+     * I render this terrain state.
+     * 
+     * Returns:
+     *  undefined.
+     */
+    render()
+    {
+        // push transformation matrix
+        push();
+
+        // rotate 60 degrees
+        rotateX(PI / 3);
+
+        // adjust sketch centering
+        translate(-this.width / 2, -this.height / 2);
+
+        // set mesh color
+        strokeWeight(0);
+        fill(...TERRAIN_COLOR);
+
+        // increase noise parameter
+        this.seed -= 0.18;

```

Comment:
Please parametrize this variable. This might be the most promising to interact with the music.